mantidproject/mantid
# Do not plot curves on a log scale where all(y)<=0

Lower the value considered as the minimum positive value for a data curve where `all(y) <= 0` is true.

**To test:**

Play with 1D plotting as much as possible. Check:

- plots where all Y data is positive for both linear & log scales 
- plots with a mix of Y data both positive & negative both linear & log scales 
- plots where all Y data is 0 on linear & log scales
- plots where all Y data is negative on linear & log scales

Also check embedded plots on instrument view & the slice viewer still work.

Fixes #19529.

[Release Notes](https://github.com/mantidproject/mantid/pull/19541/commits/c708aa33c96513a2a757b8c024d391e0c88cbe53
)

---

#### Reviewer ####

Please comment on the following ([full description](http://www.mantidproject.org/Individual_Ticket_Testing)):

##### Code Review #####


##### Functional Tests #####


- Does everything look good? Comment with the ship it emoji but don't merge. A member of `@mantidproject/gatekeepers` will take care of it.
